packages/astro/src/core/render-context.ts

```diff
@@ -463,9 +463,10 @@ export class RenderContext {
 				extraScriptHashes: [],
 				propagators: new Set(),
 			},
-			shouldInjectCspMetaTags: manifest.shouldInjectCspMetaTags,
-			clientScriptHashes: manifest.clientScriptHashes,
-			clientStyleHashes: manifest.clientStyleHashes,
+			shouldInjectCspMetaTags: manifest.csp?.shouldInjectCspMetaTags ?? false,
```

Comment:
If `shouldInjectCspMetaTags` isn't on `manifest.csp` anymore, then this flag should probably be `!!manifest.csp`

Reply:
It must stay in the manifest